qt/python/mantidqt/mantidqt/widgets/superplot/presenter.py

```diff
@@ -360,24 +371,32 @@ def _update_list(self):
                     spectra.append(data[1])
             self._view.set_spectra_list(name, spectra)
 
-    def _update_plot(self):
+    def _update_plot(self, replot=False):  # noqa C901
```

Comment:
Could complexity be reduced here to remove the need for ignoring C901?

Reply:
I have split the function in two and removed the `#noqa`

Reply:
Thank you for the split, but the `#noqa` is still I believe. Could you remove it since it's no longer necessary?

Reply:
I will force push the modification. Thanks